### ISSUE TITLE
Container failing to start on Synology NAS (DSM 7.2)

### ISSUE THREAD

**Issue by andosteinmetz** [slskd/slskd#1133]
Hi,

First of all, thanks for maintaining this project!

I just recently installed slskd on my Synology NAS and it was going great. Then, when I opened it up today I saw that it was disconnected from the Soulseek network, and wouldn't connect, so I tried restarting the container.

Since then, it fails during startup.

Here's what I'm seeing when I tail the logs:

```
[23:16:15 INF] Version: 0.20.1.0 (0.20.1.0+0e3bea64)
[23:16:15 INF] System: .NET 8.0.3, Unix 4.4.302.0, 64 bit, 2 processors
[23:16:15 INF] Process ID: 6 (64 bit)
[23:16:15 INF] Invocation ID: "2fee9a87-3e3f-4508-a717-77f6cab767aa"
[23:16:15 INF] Instance Name: default
[23:16:15 INF] Configuring application...
[23:16:15 INF] Using application directory /app
[23:16:15 INF] Using configuration file /app/slskd.yml
[23:16:15 INF] Storing application data in /app/data
[23:16:15 INF] Saving application logs to /app/logs
[23:16:17 INF] Listening for HTTP requests at http://0.0.0.0:5030/
[23:16:17 INF] Listening for HTTPS requests at https://0.0.0.0:5031/
[23:16:17 INF] Using randomly generated self-signed certificate
[23:16:17 INF] Using base url /
[23:16:17 INF] Serving static content from /slskd/wwwroot
[23:16:17 INF] Configuration complete.  Starting application...
[23:16:17 INF] Application started
[23:16:18 INF] Listening for incoming connections on 0.0.0.0:50300
[23:16:18 INF] Checking GitHub Releases for latest version
[23:16:18 INF] Initializing shares
[23:16:18 INF] Share cache StorageMode is 'Memory'. Attempting to load from backup...
[23:16:18 INF] Share cache backup validated. Attempting to restore...
[23:16:18 INF] Share cache successfully restored from backup
[23:16:18 INF] Share cache loaded from disk successfully. Sharing 2604 directories and 9408 files
[23:16:19 INF] Version 0.20.1.0 is up to date.
[23:16:21 INF] Warming browse response cache...
[23:16:21 INF] Starting system clock...
[23:16:21 INF] Browse response cached successfully in 3061ms
[23:16:21 INF] System clock started
[23:16:31 ERR] Disconnected from the Soulseek server: Operation timed out after 10000 milliseconds
[23:16:31 INF] Waiting about a second before reconnecting
[23:16:31 ERR] Hosting failed to start
System.TimeoutException: Operation timed out after 10000 milliseconds
   at Soulseek.Network.Tcp.Connection.ConnectAsync(Nullable`1 cancellationToken)
   at Soulseek.Network.Tcp.Connection.ConnectAsync(Nullable`1 cancellationToken)
   at Soulseek.Network.Tcp.Connection.ConnectAsync(Nullable`1 cancellationToken)
   at Soulseek.SoulseekClient.ConnectInternalAsync(String address, IPEndPoint ipEndPoint, String username, String password, CancellationToken cancellationToken)
   at slskd.Application.Microsoft.Extensions.Hosting.IHostedService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
   at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
[23:16:31 FTL] Application terminated unexpectedly
System.TimeoutException: Operation timed out after 10000 milliseconds
   at Soulseek.Network.Tcp.Connection.ConnectAsync(Nullable`1 cancellationToken)
   at Soulseek.Network.Tcp.Connection.ConnectAsync(Nullable`1 cancellationToken)
   at Soulseek.Network.Tcp.Connection.ConnectAsync(Nullable`1 cancellationToken)
   at Soulseek.SoulseekClient.ConnectInternalAsync(String address, IPEndPoint ipEndPoint, String username, String password, CancellationToken cancellationToken)
   at slskd.Application.Microsoft.Extensions.Hosting.IHostedService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
   at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at slskd.Program.Main(String[] args)
```

I tried deleting the image and creating a new one from scratch but that didn't seem to fix it.

I was curious if you have any idea what might have gone wrong or how I might go about repairing it.

Thanks again!

**Comment by jpdillingham** [comment]
Uncanny timing! I just hit this myself and created #1132 about 20 minutes ago to track it.  It's not a big change, and something I can get out in the next few days.

**Comment by andosteinmetz** [comment]
Ha! Fancy that. Thanks so much for the quick response

**Comment by jpdillingham** [comment]
Fix released in [0.21.0](https://github.com/slskd/slskd/releases/tag/0.21.0)
